### ISSUE TITLE
Followers do not appear when compiled with agbcc

### ISSUE THREAD

**Issue by pkmnsnfrn** [rh-hideout/pokeemerald-expansion#4938]
### Description

Reported from [The Pit](<https://discord.com/channels/1247322603895656579/1247323428814458942/1260466451517804584>)

# Videos
## 20 year old compiler
https://github.com/rh-hideout/pokeemerald-expansion/assets/77138753/f3489a0f-6594-4d53-85fc-0bf14d1068ea
## modern compiler
https://github.com/rh-hideout/pokeemerald-expansion/assets/77138753/f105e2c2-0608-449a-b644-ab2ae4634e18
# Repro
1. Start a new save file
2. Use the Cheat Start script
3. Walk outside of truck
4. Expect to see Treecko sprite behind player, but one does not appear

### Version

upcoming (Edge)

### Upcoming/master Version

b5160b7ec889eb35c2ffc2519afa3028efdce1f1

### Discord contact info

psf

**Comment by aarant** [comment]
Fixed this upstream in https://github.com/aarant/pokeemerald/commit/88cfd9e2673e462bd1f0bee058c20443ccadda3b

Basically: I was relying on EWRAM memory layout to "offset" a sprite frame's worth of data, and doing pointer arithmetic. This works fine on non-expansion builds because `gDecompressionBuffer` isn't near the start of EWRAM. But if it is, like in expansion, subtracting the offset in `LoadSpriteSheetByTemplate` will take the pointer below EWRAM and into unallocated memory, and this is what causes the gfx not to be loaded.

So, I fixed it and did it right this time, see the commit for details. Should I open a PR, or will the webhook be notification enough?
